pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPD.java

```diff
@@ -17,18 +19,21 @@
 import java.util.Set;
 import java.util.TreeMap;
 
+import org.checkerframework.checker.nullness.qual.NonNull;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import net.sourceforge.pmd.annotation.Experimental;
 import net.sourceforge.pmd.cli.internal.CliMessages;
 import net.sourceforge.pmd.lang.ast.TokenMgrError;
 import net.sourceforge.pmd.util.FileFinder;
+import net.sourceforge.pmd.util.FileUtil;
 import net.sourceforge.pmd.util.IOUtil;
 import net.sourceforge.pmd.util.database.DBMSMetadata;
 import net.sourceforge.pmd.util.database.DBURI;
 import net.sourceforge.pmd.util.database.SourceObject;
 
+@Deprecated
```

Comment:
we should probably introduce a `CpdAnalysis` similar to `PmdAnalysis`, and remove any logic from this class, to be eventually removed.